### PR TITLE
Remove spm

### PR DESCRIPTION
**Pull request by faddat**
# Description

This PR removes SPM.

## Linked issues

Closes: #753 

## Type of change

If you've checked more than one of the first three boxes, consider splitting this PR into multiple PRs!

- [ ] `Feature`: Changes and/or adds code behavior, irrelevant to bug fixes
- [x] `Fix`: Changes and/or adds code behavior, specifically to fix a bug
- [x] `Refactor`: Changes existing code style, naming, structure, etc.
- [ ] `Testing`: Adds testing
- [ ] `Docs`: Adds documentation

## Regression tests

If `Refactor`, describe the new or existing tests that verify no behavior was changed or added where refactors were introduced.

## New behavior tests

If `Feature` or `Fix`, describe the new or existing tests that verify the new behavior is correct and expected.

## Versioning Implications

- [x] This PR will affect [semantic versioning as defined for ICS](../CONTRIBUTING.md#semantic-versioning)

If the above box is checked, which version should be bumped?

- [ ] `MAJOR`: Consensus breaking changes to both the provider and consumers(s), including updates/breaking changes to IBC communication between provider and consumer(s)
- [x] `MINOR`: Consensus breaking changes which affect either only the provider or only the consumer(s)
- [ ] `PATCH`: Non consensus breaking changes

## Targeting

Please select one of the following:

- [x] This PR is only relevant to main
- [ ] This PR is relevant to main, and should also be back-ported to ____ (ex: v1.0.0 and v1.1.0)
- [ ] This PR is only relevant to ____ (ex: v1.0.0, v1.1.0, and v1.2.0)

### CODE REVIEW

**Comment by faddat** [comment]
Taints every consumer chain with known bad software

**Review by jtremback — APPROVED** [review]
LGTM

File: app/consumer-democracy/app.go [hunk 1]
```diff
@@ -903,3 +904,16 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 
 	return paramsKeeper
 }
+
+// MakeTestEncodingConfig creates an EncodingConfig for testing. This function
+// should be used only in tests or when creating a new app instance (NewApp*()).
+// App user shouldn't create new codecs - use the app.AppCodec instead.
+// [DEPRECATED]
```
Comment (smarshall-spitzbart) [inline app/consumer-democracy/app.go:911]: Why does this PR add functions that are marked as depreciated? Did this code already exist in some form?

File: app/params/doc.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+Package params defines the simulation parameters in the gaia.
```
Comment (smarshall-spitzbart) [inline app/params/doc.go:2]: Are these comments still relevant to the ICS repo?

File: app/provider/app.go [hunk 3]
```diff
@@ -875,3 +875,16 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 
 	return paramsKeeper
 }
+
+// MakeTestEncodingConfig creates an EncodingConfig for testing. This function
+// should be used only in tests or when creating a new app instance (NewApp*()).
+// App user shouldn't create new codecs - use the app.AppCodec instead.
+// [DEPRECATED]
```
Comment (smarshall-spitzbart) [inline app/provider/app.go:882]: Same question as above

File: app/consumer/app.go [hunk 4]
```diff
@@ -766,3 +766,16 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 
 	return paramsKeeper
 }
+
+// MakeTestEncodingConfig creates an EncodingConfig for testing. This function
+// should be used only in tests or when creating a new app instance (NewApp*()).
+// App user shouldn't create new codecs - use the app.AppCodec instead.
+// [DEPRECATED]
```
Comment (smarshall-spitzbart) [inline app/consumer/app.go:773]: same question as above

**Review by smarshall-spitzbart — APPROVED** [review]
Couple of clarification questions but nothing seemed dangerous, LGTM

**Comment by faddat** [comment]
the pull request removes spm, which is dangerous thanks

File: app/consumer-democracy/app.go (hunk 1 above)
Comment (faddat) [reply app/consumer-democracy/app.go:911]: I copied it out of the last SPM removal I did with my own two hands.

Comment (faddat) [reply app/consumer-democracy/app.go:911]: but -- to be clear, this is a deprecated way to do things and we have been trying to pull this out of the sdk for ages.

File: app/params/doc.go (hunk 2 above)
Comment (faddat) [reply app/params/doc.go:2]: here, let's improve all this a bit

File: app/consumer/app.go (hunk 4 above)
Comment (faddat) [reply app/consumer/app.go:773]: this is a shim due to the design of ics

File: app/provider/app.go (hunk 3 above)
Comment (faddat) [reply app/provider/app.go:882]: this is a shim due to the design of ics

File: app/consumer-democracy/app.go (hunk 1 above)
Comment (faddat) [reply app/consumer-democracy/app.go:944]: this is a shim due to the design of ics

**Comment by faddat** [comment]
the code exists in the params folder too.  For some reason it did not work when I tried that.  So I put it in app.go, kinda same as Marius did

**Comment by faddat** [comment]
but @smarshall-spitzbart sir you are correct that is a really unusual place for those things!

**Comment by faddat** [comment]
ah right that's what happened.

I prefer the prior version.  Tests work.

**Comment by faddat** [comment]
Interestingly, even Marius's own team noted the bug that was copied into the SPSP branch
